lib/stale.js

```diff
@@ -10,28 +10,54 @@ module.exports = class Stale {
 
     await this.ensureStaleLabelExists();
 
-    this.github.paginate(this.getStaleIssues(), data => {
-      data.items.filter(issue => !issue.locked)
-                .forEach(issue => this.mark(issue));
-    });
+    this.mark();
+    this.sweep();
+  }
+
+  mark() {
+    const staleDays = this.config.days || this.config.daysUntilStale;
+
+    var markedIssues = new Set()
+
+    this.getStaleIssuesSearches(staleDays).forEach(search =>
+      this.github.paginate(search, data => {
+        data.items.filter(issue => !issue.locked)
+                  .filter(issue => !markedIssues.includes(issue))
+                  .forEach(issue => this.mark(issue))
+                  .forEach(issue => markedIssues.add(issue));
+      })
+    );
+  }
 
-    this.github.paginate(this.getClosableIssues(), data => {
+  sweep() {
+    const closeDays = this.config.days || this.config.daysUntilClose;
+
+    this.github.paginate(this.getClosableIssuesSearch(), data => {
       data.items.filter(issue => !issue.locked)
                 .forEach(issue => this.close(issue));
     });
   }
 
-  getStaleIssues() {
-    const labels = [this.config.staleLabel].concat(this.config.exemptLabels);
-    const query = labels.map(label => `-label:"${label}"`).join(' ');
-    const days = this.config.days || this.config.daysUntilStale;
-    return this.search(days, query);
+  getStaleIssuesSearches(staleDays) {
+    const onlyLabels = this.config.onlyLabels;
+
+    if (onlyLabels.length == 0) {
+      const labels = [this.config.staleLabel].concat(this.config.exemptLabels);
+      const query = labels.map(label => `-label:"${label}"`).join(' ');
+      return [this.search(staleDays, query)];
+    } else {
+      return onlyLabels.map(label => this.getStaleIssuesSearch(staleDays, label));
+    }
+  }
+
+  getStaleIssuesSearch(staleDays, label) {
```

Comment:
Returns a search that should discover all issues tagged with the given label that are not also tagged with the stale label.